### PR TITLE
New issue: basic_osyncstream::rdbuf needs a const_cast

### PR DESCRIPTION
**Pull request by timsong-cpp**

### CODE REVIEW

File: xml/issue3127.xml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+namespace std {
+  template&lt;class charT, class traits, class Allocator&gt;
+  class basic_osyncstream : public basic_ostream&lt;charT, traits&gt; {
+  public:
+    [&hellip;]
+
+    // <sref ref="[syncstream.osyncstream.members]"/>, member functions
+    void emit();
+    streambuf_type* get_wrapped() const noexcept;
+    syncbuf_type* rdbuf() const noexcept { return <ins>const_cast&lt;syncbuf_type *&gt;(</ins>&amp;sb<ins>)</ins>; }
```
Comment (CaseyCarter) [inline xml/issue3127.xml]: We should drive-by change this `&` to `addressof`: `syncbuf_type` is `basic_syncbuf<charT, traits, Allocator>` and therefore subject to template-parameter ADL hijacking.

Comment (timsong-cpp) [reply xml/issue3127.xml]: Yes, I want to do that separately. There are 27 instances of `&sb` in Clause 30 alone that probably all need to be turned into `addressof`, and potentially other things too - I haven't done a thorough audit.

Comment (timsong-cpp) [reply xml/issue3127.xml]: See #211.
